modo/cram.py

```diff
@@ -3,16 +3,51 @@
 from pysam import AlignmentFile, AlignmentHeader
 from rdflib import Graph
 
+# MODO.slice(data: str, coords: str) will call slice(), which in turn will
+# call the right slicing function depending on the data format/type that
+# is being sliced (e.g., if it is a CRAM file, it will call slice_cram())
 
-def slice(cram_path: AlignmentFile, coords: str) -> AlignmentFile:
-    """Return a slice of the CRAM File.
 
-    Examples
-    --------
-    >>> slice("data/ex1/demo1.cram", "chr1:100-200")
+def slice(data_name: str, data_path: str, coords: str):
+    "Returns a slice of the requested region for the requested omics type"
+
+    # get the data type (e.g., CRAM, array) of the requested data
+    # data_type =    # Cyril, please add in the logic to get the data type
+    # data_path =    # and the data_path from the metadata
+
+    if data_type == "cram":
+        return slice_cram(data_path, coords)
+    elif data_type == "array":
+        return slice_array(data_path, coords)  # To be added after we know
+        # what this data looks like
+
+
+def slice_cram(cram_path: AlignmentFile, coords: str):  # -> AlignmentFile:
+    """Return a slice of the CRAM File as an iterator object.
+
+    Usage:
+    -----
+    >>> x = slice("data/ex1/demo1.cram", "chr1:100-200")
+    >>> for read in x:
+            print(read)
     """
-    # https://htsget.readthedocs.io/en/stable/index.html
-    ...
+
+    # split up coordinate string "chr:start-end" into its three elements
+    coords = coords.replace("-", ":")
```

Comment:
**fix:** this is dangerous, as it is possible for a chromosome name to contain `-`. This would break in such cases.